### DISCUSSION TITLE
Stripe integration?

### DISCUSSION THREAD

**Discussion by richardpickett** [docker-library/ghost#366] [General]
If I'm reading the docs right, in order for ghost to receive updates from stripe, it has to run the stripe cli, which forwards the requests on to ghost.

[Ghost Documentation: Stripe Webhook](https://ghost.org/docs/webhooks/#stripe-webhooks)

Does stripe integration work with the docker-based ghost deploy? (I'm also not seeing Stripe in the ghost settings, so I'm guessing "no")

If not, this may be a problem I'd be interested in solving, caveat: my solution will be to have two running programs, first starting the stripe cli and feeding the webhook secret into the cmdline ghost start (two programs in one container is antithetical to the container pattern)

**Comment by richardpickett** [comment]
I now have this working.

On start first the stripe client is started in the background and the script waits until stripe console outputs the webhook secret, which it then sets as an env variable and starts ghost.

lmk if you'd like me to submit a PR with the changes - if you think it'd be useful for your codebase.

Pros:
1. Stripe console runs on the same container as ghost.
2. The webhook is automatically passed to ghost.  If you run them separately, every time the console restarts you have to get the new webhook secret, set it in the ghost environment, and then restart ghost.  No need with my way.

Cons:
1. It breaks the "one container, one process" paradigm.  (I think worth it in this case)
2. In the future if the ghost team enables multiple ghost containers to serve the same site, it will be necessary to set one and only one instance to run the stripe console.  Easily achieved with two simultaneous deployments - one runs a single instance of stripe+ghost, the other scales multiple instances of ghost as needed. If and when the multiple container feature is enabled, I'll revisit my solution and enable an environment flag to enable/disable stripe, allowing multi-deployment.

**Reply by yosifkit** [reply]
Unfortunately a backgrounded process would not be accepted into official images. Since there is nothing monitoring that when it might crash, and the container could be left in a half working state. We've looked at many of the `init`-like replacements for running multiple processes in containers and have not found one the fits all the needs.

**Reply by richardpickett** [reply]
Au contraire mon frère - I have a script that monitors that both `stripe listen` and `node content/index.js` are both running.  When either ends, the script restarts as appropriate (if stripe exits, it gets the updated token then restarts node; if it's node that exited, it just restarts node).

So your "since there is nothing monitoring..." statement is wrong.

If you go the `init` route, you still have to build in the complexity that when stripe cycles you also cycle node - which is already handled by my script.
